### ISSUE TITLE
AttributeError in trees/binarysearchtree.py

### ISSUE THREAD

**Issue by ayush1997** [prakhar1989/Algorithms#46]
```
a = BinarySearchTree()
a.insert(12)
a.insert(1)
a.insert(10)
```

raises error
```
Traceback (most recent call last):
  File "binarysearchtree.py", line 147, in <module>
    a.insert(1)
  File "binarysearchtree.py", line 109, in insert
    if node.value == value:
AttributeError: 'NoneType' object has no attribute 'value'
```

**Comment by ayush1997** [comment]
Please review PR#47 for this issue.

**Comment by prakhar1989** [comment]
Closed in #47
